devito/types/sparse.py

```diff
@@ -62,8 +66,24 @@ def __init_finalize__(self, *args, **kwargs):
     @classmethod
     def __indices_setup__(cls, *args, **kwargs):
         dimensions = as_tuple(kwargs.get('dimensions'))
+        # Need this not to break MatrixSparseFunction
+        try:
+            _sub_funcs = tuple(cls._sub_functions)
+        except TypeError:
+            _sub_funcs = ()
+        # If a subfunction provided use the sparse dimension
+        for f in _sub_funcs:
+            if f in kwargs:
+                try:
+                    sparse_dim = kwargs[f].indices[0]
+                    break
+                except AttributeError:
+                    continue
+        else:
```

Comment:
Wrong indent level for this `else`?

Reply:
No the `else` goes there with the `for`. It's a python construct "if for didn't exit then " that avoids wrapping the for in a try/if

Reply:
Huh, TIL

Reply:
yes and it's quite handy